src/api.ts

```diff
@@ -0,0 +1,41 @@
+import { TokenEndpointOptions } from './global';
+import { DEFAULT_AUTH0_CLIENT } from './constants';
+import { getJSON } from './http';
+
+export type TokenEndpointResponse = {
+  id_token: string;
+  access_token: string;
+  refresh_token?: string;
+  expires_in: number;
+  scope?: string;
+};
+
+export async function oauthToken(
+  {
+    baseUrl,
+    timeout,
+    audience,
+    scope,
+    auth0Client,
+    ...options
+  }: TokenEndpointOptions,
+  worker?: Worker
+): Promise<TokenEndpointResponse> {
+  return await getJSON(
```

Comment:
I do not feel strong about this, but just wanted to ask. Is there any reason we are returning `Promise<any>` from getJSON instead of making it generic ? I am not sure we gain much other then removing a couple of unnecessary `any`.